test/execution/test_protocol.py

```diff
@@ -231,6 +231,28 @@ def protocol(roles):
     assert (dict_res["alice"][0] == th.tensor([4])).all()
 
 
+def test_stateful_protocol(workers):
+    shapes = {"alice": ((1,),), "bob": ((1,),)}
+    states = {"alice": (th.tensor([1]), th.tensor([3])), "bob": (th.tensor([5]),)}
+
+    @sy.func2protocol(args_shape=shapes, states=states)
+    def protocol(roles):
+        # fetch tensors from stores
+        # TODO will we fetch tensors that come from states this way?
+        # With tags?
```

Comment:
🤔 

For `Plans`, they're fetched with `state.read()`, right? I remember you saying you weren't a fan of that, so maybe this is a good time to think about how to do it better.

Reply:
Yep, they're fetched with `read`. As we are going towards something very similar for other tensors fetched from the worker store, I think we'll need to talk about differences between them/if we can totally merge these 2 concepts (from an API point of view at least).

Reply:
Maybe in the Syft Core sync tomorrow?